### DISCUSSION TITLE
How do I correctly use Python 3.9?

### DISCUSSION THREAD

**Discussion by simon-friedberger** [Homebrew/discussions#5196] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.12
ORIGIN: https://github.com/Homebrew/brew
HEAD: 780fbbc65e90fbe09629aba180a1839e9e7dbaf2
Last commit: 23 hours ago
Core tap JSON: 12 Mar 13:47 UTC
Core cask tap JSON: 12 Mar 13:47 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.44.0 => /opt/homebrew/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.4-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.3
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  mitmproxy
  python-mutagen

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  dog
  exa
  openssl@1.1
  python@3.8

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  python-packaging
  ipython
```

### Description of issue

My path includes (in that order):
* /opt/homebrew/bin
* /opt/homebrew/opt/python@3.9/libexec/bin

`/opt/homebrew/bin` has `python3` pointing to Python 3.12. `/opt/homebrew/opt/python@3.9/libexec/bin` has python and python3.

This leads to the confusing situation that calling python gives 3.9 and calling python3 gives 3.12.
Is this intentional? Is an intended order for the homebrew additions to PATH?

Modifying the path is suggested e.g. in this answer: https://github.com/orgs/Homebrew/discussions/5154#discussioncomment-8564047 but I couldn't find anything straightforward here: https://docs.brew.sh/Homebrew-and-Python

So, in general, what is the recommended way to switch the default python to a specific (older) version?

**Comment by SMillerDev** [comment]
> So, in general, what is the recommended way to switch the default python to a specific (older) version?

Use a tool like https://github.com/pyenv/pyenv?tab=readme-ov-file#homebrew-in-macos, Homebrew's tools work best if you do not have specific version needs.

**Comment by branchv** [comment]
> My path includes (in that order):

Just reverse the order so 3.9 takes precedence:

```bash
$ export PATH="/opt/homebrew/opt/python@3.9/libexec/bin:/opt/homebrew/bin:$PATH"
$ which python python3 python3.12
/opt/homebrew/opt/python@3.11/libexec/bin/python
/opt/homebrew/opt/python@3.11/libexec/bin/python3
/opt/homebrew/bin/python3.12
 ```
